linkerd/app/inbound/src/policy/store.rs

```diff
@@ -31,51 +27,46 @@ struct PortHasher(u16);
 
 // === impl Store ===
 
-impl Store {
-    fn mk_default(default: DefaultPolicy) -> Option<(Tx, Rx)> {
-        match default {
-            DefaultPolicy::Deny => None,
-            DefaultPolicy::Allow(sp) => Some(watch::channel(sp)),
-        }
-    }
-
-    pub(crate) fn fixed(
-        default: impl Into<DefaultPolicy>,
+impl<S> Store<S> {
+    pub(crate) fn spawn_fixed(
+        default: DefaultPolicy,
         ports: impl IntoIterator<Item = (u16, ServerPolicy)>,
-    ) -> (Self, Option<Tx>) {
-        let rxs = ports
-            .into_iter()
-            .map(|(p, s)| {
+    ) -> Self {
+        let cache = {
+            let rxs = ports.into_iter().map(|(p, s)| {
                 // When using a fixed policy, we don't need to watch for changes. It's
                 // safe to discard the sender, as the receiver will continue to let us
                 // borrow/clone each fixed policy.
                 let (_, rx) = watch::channel(s);
                 (p, rx)
-            })
-            .collect();
-
-        let (default_tx, default) = match Self::mk_default(default.into()) {
-            Some((tx, rx)) => (Some(tx), Some(rx)),
-            None => (None, None),
+            });
+            Cache::with_permanent_from_iter(Duration::MAX, rxs)
         };
 
-        let store = Self {
-            default,
-            ports: Arc::new(rxs),
-        };
-        (store, default_tx)
+        let (default_tx, default_rx) = watch::channel(ServerPolicy::from(default));
+        tokio::spawn(async move {
+            default_tx.closed().await;
+        });
```

Comment:
maybe worth a comment here saying that this task's job is to keep the watch from closing if the tx is dropped?